### PR TITLE
Throw an exception for Object children without @objc properties.

### PR DESCRIPTION
**Pull request by pavel-ship-it**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@ x.y.z Release notes (yyyy-MM-dd)
 =============================================================
 ### Enhancements
 * Add the ability to listen for when a Watch Change Stream is opened when using Combine. Use `onOpen(event:)` directly after opening a `WatchPublisher` to register a callback to be invoked once the change stream is opened.
+* Throw an exception for Object children without @objc properties.
```
Comment (leemaguire) [inline CHANGELOG.md:5]:
```suggestion
* Throw an exception for Objects that have none of its properties marked with @objc.
```

File: Realm/RLMSchema.mm [hunk 2]
```diff
@@ -95,6 +95,9 @@ @implementation RLMSchema {
         s_sharedSchema.objectSchemaByName[schema.className] = schema;
     }
 
+    if ([cls shouldIncludeInDefaultSchema] && schema.isSwiftClass && schema.properties.count == 0) {
+        @throw RLMException(@"No properties was defined for '%@'. Did you remember to add '@obj' in your model?", schema.className);
```
Comment (leemaguire) [inline Realm/RLMSchema.mm:99]:
```suggestion
        @throw RLMException(@"No properties are defined for '%@'. Did you remember to mark them with '@obj' in your model?", schema.className);
```

**Review by leemaguire — CHANGES_REQUESTED** [review]
This needs a test

**Comment by pavel-ship-it** [comment]
There is a problem @leemaguire - it will throw an exception before any test will start it's cycle.

**Comment by tgoyne** [comment]
You can look at SchemaTests.mm to see examples of how we test schema errors. The basic idea is that they inherit from a different base class which won't be included in the default schema.

File: Realm/RLMSchema.mm [hunk 3]
```diff
@@ -95,6 +95,9 @@ @implementation RLMSchema {
         s_sharedSchema.objectSchemaByName[schema.className] = schema;
     }
 
+    if ([cls shouldIncludeInDefaultSchema] && schema.isSwiftClass && schema.properties.count == 0) {
```
Comment (tgoyne) [inline Realm/RLMSchema.mm:98]: This isn't the right place for this check. It should go with the existing validation logic in RLMObjectSchema.mm.

File: Realm/RLMObjectSchema.mm [hunk 4]
```diff
@@ -173,6 +173,9 @@ + (instancetype)schemaForObjectClass:(Class)objectClass {
         }
     }
 
+    if ([objectClass shouldIncludeInDefaultSchema] && schema.isSwiftClass && schema.properties.count == 0) {
+        @throw RLMException(@"No properties are defined for '%@'. Did you remember to mark them with '@obj' in your model?", schema.className);
```
Comment (tgoyne) [inline Realm/RLMObjectSchema.mm:177]: Missing the 'c' in `@objc`

**Review by leemaguire — APPROVED** [review]

**Review by tgoyne — APPROVED** [review]

**Pull request merged**